INT-1618 Module to Wrap JIRA API

shum opened the pull request:
Implemented functionality to query JIRA API for issues based on JQL provided, implemented testing

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mongodb-js/jira-api-wrapper/1)

<!-- Reviewable:end -->

shum committed:
WIP initial working protype of JIRA API wrapper, able to search with JQL

shum committed:
searchJQL changed to getIssuesJQL to follow method naming convention

shum committed:
add .env to gitignore, use local env variables instead

shum committed:
testing added

imlucas commented:
Thanks! This is shaping up nicely.  A few suggested updates before we publish this to npm are below.

---

Reviewed 12 of 13 files at r1.
Review status: 12 of 13 files reviewed at latest revision, 4 unresolved discussions.

---

_[package.json, line 2 [r1]](https://reviewable.io:443/reviews/mongodb-js/jira-api-wrapper/1#-KMVsEHpfWfUJfYJVKSD:-KMVsEHqjwG-DOd4Cc-c:-1012281112) ([raw file](https://github.com/mongodb-js/jira-api-wrapper/blob/09c8c9012eb6a25754346c5ceccee05a329aa7d3/package.json#L2)):_

> ``` JSON
> {
>   "name": "jira-api-wrapper",
> ```

Let's rename this module `jira-issue-read-stream` before publishing.

---

_[lib/index.js, line 53 [r1]](https://reviewable.io:443/reviews/mongodb-js/jira-api-wrapper/1#-KMVsg9Se4y8sPF2vDxG:-KMVsg9Se4y8sPF2vDxH:440024601) ([raw file](https://github.com/mongodb-js/jira-api-wrapper/blob/09c8c9012eb6a25754346c5ceccee05a329aa7d3/lib/index.js#L53)):_

> ``` JavaScript
> };
> 
> var getIssuesJQL = function(jql, callback) {
> ```

Just make this your `module.exports = function(jql, callback)`

---

_[lib/index.js, line 77 [r1]](https://reviewable.io:443/reviews/mongodb-js/jira-api-wrapper/1#-KMVt-zA5E7RUMoXlkuW:-KMVt-zA5E7RUMoXlkuX:1641399307) ([raw file](https://github.com/mongodb-js/jira-api-wrapper/blob/09c8c9012eb6a25754346c5ceccee05a329aa7d3/lib/index.js#L77)):_

> ``` JavaScript
>     }
> 
>     // res.on('data', function(data) {
> ```

Please don't commit commented out code.  I'd recommend the http://npm.im/debug module instead.

---

_[test/testVars.js, line 3 [r1]](https://reviewable.io:443/reviews/mongodb-js/jira-api-wrapper/1#-KMVt_-awGf--NF1E5af:-KMVt_-awGf--NF1E5ag:2031639036) ([raw file](https://github.com/mongodb-js/jira-api-wrapper/blob/09c8c9012eb6a25754346c5ceccee05a329aa7d3/test/testVars.js#L3)):_

> ``` JavaScript
> require('dotenv').config();
> 
> var URL = 'https://jira.mongodb.org/rest/api/2';
> ```

move to an ​environment variable

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/mongodb-js/jira-api-wrapper/1#-:-KMVtmt5j0fQ6BBBd_Al:-1879065857)_

<!-- Sent from Reviewable.io -->

imlucas commented:
Reviewed 1 of 13 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/mongodb-js/jira-api-wrapper/1)_

<!-- Sent from Reviewable.io -->

shum committed:
lib: reformat module.exports so that it only exports one function, remove commented out code
test: move URL to env variable, modify to import lib method as one function
package.json: rename module to jira-issue-read-stream

imlucas commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/mongodb-js/jira-api-wrapper/1#-:-KMj0unNXquWG_ooiiim:1417088721)_

<!-- Sent from Reviewable.io -->
